### PR TITLE
fix: encoder_hidden_​​state_mask is ignored in Qwen-Image attention

### LINKED ISSUE #686: potential Qwen Image batching issue
Hi @kohya-ss 
I haven't confirmed - still downloading the models and learning how to set everything up, because I want to try Qwen Image Edit training - but it looks like your code is affected by this bug:
https://github.com/huggingface/diffusers/issues/12294

It can lead to incorrect conditioning if batch size > 1
A workaround is posted in the `diffusers` issue. Or, since you've copied the pipeline into your repo, it could also be fixed there directly.

### PR DESCRIPTION
**Pull request by kohya-ss**
This only happened when `--split_attn` was not specified (in SDPA), not when `--split_attn` was specified (attention mechnism other than SDPA requires `--split_attn` for now.

Closes #686 .

### CODE REVIEW

File: src/musubi_tuner/qwen_image/qwen_image_model.py [hunk 1]
```diff
@@ -727,6 +727,23 @@ def forward(
...
+            if encoder_hidden_states_mask is not None:
+                # encoder_hidden_states_mask: [B, S_txt]
+                attention_mask = torch.cat(
+                    [
+                        torch.ones(
+                            (encoder_hidden_states_mask.shape[0], seq_img), device=encoder_hidden_states_mask.device, dtype=torch.bool
+                        ),
+                        encoder_hidden_states_mask.to(torch.bool)
+                    ]
+                    , dim=1
```
[the remark this answers is not in the document]
Comment (kohya-ss) [reply src/musubi_tuner/qwen_image/qwen_image_model.py:741]: Well, but this is formatted by Ruff.

**Comment by FurkanGozukara** [comment]
Does this fix requires re-training of batch size 1? as far as i understood it fixed for batch size 2 or bigger right?

Oh I was using --split_attn as well

**Comment by kohya-ss** [comment]
There will be no impact if `--split_attn` is specified.

If `--split_attn` is not specified, the mask is not applied, but the corresponding part of encoder_hidden_states has a value of 0, so I don't think it will have much of an impact on training.
